### PR TITLE
RUBY-1868 Assorted fixes to improve test suite reliability

### PR DESCRIPTION
**Pull request by p-mongo**

### CODE REVIEW

**Comment by p-mongo** [comment]
This hasn't gotten Evergreen to green quite yet, unfortunately.

File: spec/mongo/cursor_spec.rb [hunk 1]
```diff
@@ -273,7 +279,7 @@
     context 'when the cursor is not fully iterated and is garbage collected' do
 
       let(:documents) do
-        (1..4).map{ |i| { field: "test#{i}" }}
+        (1..6).map{ |i| { field: "test#{i}" }}
```
Comment (saghm) [inline spec/mongo/cursor_spec.rb]: It's not clear at all to me why we made 3 documents before or why having 5 now is better. Why does this need to be changed?

File: lib/mongo/cluster.rb [hunk 2]
```diff
@@ -150,12 +160,14 @@ def initialize(seeds, monitoring, options = Options::Redacted.new)
...
 
-      @cursor_reaper = CursorReaper.new
-      @socket_reaper = SocketReaper.new(self)
-      @periodic_executor = PeriodicExecutor.new(@cursor_reaper, @socket_reaper)
-      @periodic_executor.run!
+      if options[:cleanup] != false
+        @cursor_reaper = CursorReaper.new
+        @socket_reaper = SocketReaper.new(self)
+        @periodic_executor = PeriodicExecutor.new(@cursor_reaper, @socket_reaper)
+        @periodic_executor.run!
```
[the remark this answers is not in the document]
Comment (p-mongo) [reply lib/mongo/cluster.rb]: Yes it should I think, but let's make that change in a separate PR.

File: spec/mongo/cursor_spec.rb (hunk 1 above)
Comment (p-mongo) [reply spec/mongo/cursor_spec.rb]: 6 total documents mean the collection requires two getmores to iterate after the first getmore is done, which is needed for jruby as described in the comment in the next hunk.

Comment (saghm) [reply spec/mongo/cursor_spec.rb]: Ah, okay. I hadn't realized that this change was also related to that comment.

**Pull request merged**